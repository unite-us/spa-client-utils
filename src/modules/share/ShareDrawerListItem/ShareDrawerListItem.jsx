import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const ShareDrawerListItem = ({ children, className }) => (
  <li className={classNames('share-drawer-list-item', className)}>
    {children}
  </li>
);

ShareDrawerListItem.propTypes = {
  /** class name */
  className: PropTypes.string,
  /** list item content */
  children: PropTypes.node.isRequired,
};

ShareDrawerListItem.defaultProps = {
  className: '',
};

export default ShareDrawerListItem;
