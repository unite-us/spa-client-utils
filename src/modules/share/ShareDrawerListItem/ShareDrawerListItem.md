```jsx
<ShareDrawerListItem
  className="this-share-drawer-list-item"
>
  A Single Share Drawer List Item
</ShareDrawerListItem>
```
