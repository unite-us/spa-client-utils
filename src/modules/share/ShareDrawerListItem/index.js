import ShareDrawerListItem from './ShareDrawerListItem';

export default ShareDrawerListItem;
