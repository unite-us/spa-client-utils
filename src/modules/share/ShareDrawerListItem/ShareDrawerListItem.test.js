import React from 'react';
import renderer from 'react-test-renderer';
import ShareDrawerListItem from './ShareDrawerListItem';

describe('ShareDrawerListItem', () => {
  it('renders list item with content', () => {
    expect(renderer.create(
      <ShareDrawerListItem className="additional-class">
        Share drawer list item
      </ShareDrawerListItem>,
    )).toMatchSnapshot();
  });
});
