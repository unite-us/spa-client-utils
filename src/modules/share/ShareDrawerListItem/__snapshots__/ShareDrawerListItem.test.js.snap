// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareDrawerListItem renders list item with content 1`] = `
<li
  className="share-drawer-list-item additional-class"
>
  Share drawer list item
</li>
`;
